stock_picking_invoicing/wizards/stock_invoice_onshipping.py

```diff
@@ -338,13 +338,18 @@ def _build_invoice_values_from_pickings(self, pickings):
             'partner_id': partner_id,
             'account_id': account_id,
             'payment_term_id': payment_term,
+            'purchase_id': picking.purchase_id.id or False,
```

Comment:
but of course if you go for the if hasattr(picking, 'purchase_id'), then you should remove this line with picking.purchase_id.id or you will have an error...